src/main/c/ssl.c

```diff
@@ -231,26 +231,35 @@ static const jint supported_ssl_opts = 0
 
 static int ssl_tmp_key_init_rsa(int bits, int idx)
 {
-#if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(OPENSSL_USE_DEPRECATED)
-    if (!(SSL_temp_keys[idx] =
-          RSA_generate_key(bits, RSA_F4, NULL, NULL))) {
-#ifdef OPENSSL_FIPS
-        /**
-         * With FIPS mode short RSA keys cannot be
-         * generated.
-         */
-        if (bits < 1024)
-            return 0;
-        else
-#endif
-        return 1;
-    }
-    else {
+#if defined(OPENSSL_FIPS)
```

Comment:
I don't have much context here but I see in the 1_0_2 and 1_0_1 branch version of `RSA_generate_key_ex` also has some `OPENSSL_FIPS` macros (see [rsa_gen.c](https://github.com/openssl/openssl/blob/OpenSSL_1_0_2-stable/crypto/rsa/rsa_gen.c#L88)). Do we still need this special case? It looks like 1_0_0 doesn't have these defines though ... so maybe safer to keep these defines (if we are sure they are correct).